### PR TITLE
Throw exception if Hive thinks it's running in local mode

### PR DESCRIPTION
**Pull request by pwendell**
This throws a descriptive error message if a user tries to run Shark in a way that will result in query responses being silently dropped.

### CODE REVIEW

File: src/main/scala/shark/execution/FileSinkOperator.scala [hunk 1]
```diff
@@ -119,6 +120,18 @@ class FileSinkOperator extends TerminalOperator with Serializable {
     val inputRdd = if (parentOperators.size == 1) executeParents().head._2 else null
     val rdd = preprocessRdd(inputRdd)
 
+    val master = rdd.context.master
+    val sparkIsLocal = (master == "local" || master.startsWith("local["))
```
Comment (rxin) [inline src/main/scala/shark/execution/FileSinkOperator.scala]: Also allow local-cluster[

**Comment by rxin** [comment]
To make this more robust, Patrick is going to expose the isLocal attribute in SparkContext and update this pull request.

**Comment by pwendell** [comment]
Jenkins, test this please.

**Comment by pwendell** [comment]
Hey @rxin - this should be good to go now. Mind taking a look and/or merging?

**Comment by rxin** [comment]
Thanks, Patrick!

**Pull request merged**